scripts/content.js

```diff
@@ -24,7 +24,7 @@ fetch(proxyurl + url) // https://cors-anywhere.herokuapp.com/https://example.com
     brands = resp;
     const finalResult = brands.result.map((data)=>{
     const result =   webbrands.filter((wellspentBrand)=>{
-        return wellspentBrand===data.BrandName
+        return wellspentBrand.includes(data.BrandName.toLowerCase())
       })
       return result;
```

Comment:
instead of return result, you need to return just the brandname (in an array):

      `return result.length > 0 ? [data.BrandName.toLowerCase()] : [];`

Reply:
This is the main fix, mentioned in my previous comment. Please make sure you understand what is happening here, I can try to explain if you don't, but try to read and understand the code (you wrote it!)

Reply:
After returning the whole data you have to filter the result to get rid of the empty arrays 
```js
 const filteredResult=  finalResult.filter((item)=>{
      return item.length >0
    }) ``` 

After that, you don't need to use ```Promise.all()``` because you used filter inside of the map, just use the ```filteredResult``` straight. you will get the result that you want!

Reply:
@m4v15 we will get another problem by using ```includes``` which is getting a substring and considering it as a brand name like ```ted``` from the world ```printed``` in this sentence ```ASOS SUPPLY printed pen pot```, I think we should include a space before the world!